spec/spec.md

```diff
@@ -364,153 +353,18 @@ values, and an explanation why a certain item or set of data is being requested:
-
-    - The _constraints object_ ****MAY**** contain a `same_subject` property. If
-      present, its value ****MUST**** be an array of objects composed as
-      follows:
-        - The _same-subject object_ ****MUST**** contain a `field_id` property.
-          The value of this property ****MUST**** be an array of strings, with
-          each string matching the string value from a _field object_'s `id`
-          property. This identifies the attributes whose [[Ref:Subject]] is of
-          concern to the [[ref:Verifier]]. It is important to note that the
-          attributes whose [[Ref:Subject]] is of concern to the [[ref:Verifier]]
-          ****MAY**** be identified in the _field object_ of a different
-          [[ref:Input Descriptor Object]].
-        - The _same-subject object_ ****MUST**** contain a `directive` property.
-          The value of this property ****MUST****  be one of the following
-          strings:
-            - `required` - This indicates that the processing entity
-              ****MUST**** include proof that the [[Ref:Subject]] of each
-              attribute identified by a value in the `field_id` array is the
-              same as the [[Ref:Subject]] of the attributes identified by the
-              other values in the `field_id` array.
-            - `preferred` - This indicates that it is ****RECOMMENDED**** that
-              the processing entity include proof that the [[Ref:Subject]] of
-              each attribute identified by a value in the `field_id` array is
-              the same as the [[Ref:Subject]] of the attributes identified by
-              the other values in the `field_id` array.
-
-      The `same_subject` property would be used by a [[ref:Verifier]] to require
-      that certain provided inputs be about the same [[Ref:Subject]]. For
-      example, a [[ref:Presentation Definition]] might contain an
-      [[ref:Input Descriptor]] which calls for a street address from a driver
-      license [[ref:Claim]] and another [[ref:Input Descriptor]] which calls
-      for a name from a birth certificate [[ref:Claim]]. Using the
-      `same_subject` property, [[ref:Verifier]] would be able to require that
-      the [[Ref:Subject]] of the street address attribute [[ref:Claim]] is the
-      same as the [[Ref:Subject]] of the name attribute.
+    - The _constraints objext_ ****MAY**** contain a `types` property. If present it's value ****MUST**** be an array of objects as composed as follows:
+      - The _types object_ ****MUST**** contains a `path` property. The value of this property has the same requirements as the _fields object_'s `path` property
+      - The _types object_ ****MUST**** contain a `filter` property. The value of this property has the same requirements as the _fields object_'s `filter` property
+      - The _types object_ ****MAY**** contain an `id` property. The value of this property has the same requirements as the _fields object_'s `id` property
```

Comment:
The reason the `fields` object has an `id` property is because it is referred to elsewhere in the spec. Is there a similar requirement here?